omod/src/main/java/org/openmrs/module/fhir2/web/servlet/FhirRestServlet.java

```diff
@@ -92,4 +97,9 @@ public void setResourceProviders(Collection<IResourceProvider> theProviders) {
 	public void setServerAddressStrategy(IServerAddressStrategy theServerAddressStrategy) {
 		super.setServerAddressStrategy(theServerAddressStrategy);
 	}
+	
+	protected void setDefaultNarrativeGenerator() {
+		this.setFhirContext(FhirContext.forR4());
+		getFhirContext().setNarrativeGenerator(new CustomThymeleafNarrativeGenerator(hapiDefaultNarrativesPropFile));
+	}
```

Comment:
Please don't do this this way. The problem here is that you're hard-coding the `FhirContext` which is actually being set by Spring (see above). The correct thing to do here would be to replace where you have the call to `this.setDefaultNarrativeGenerator()` with the last line of code:

```java
getFhirContext().setNarrativeGenerator(new CustomThymeleafNarrativeGenerator(hapiDefaultNarrativesPropFile));
```

That way, it only depends on the FhirContext that is set by Spring, so that we don't break support for DSTU3 while doing this.

It might also be worth thinking about if we could load the `ThymeleafNarrativeGenerator` from Spring, as this might give us more flexibility in defining the property files to load, instead of needing to know about them ahead of time.

Reply:
Okay, I'll do the changes.